### ISSUE TITLE
Large jump in display size when QT_FONT_DPI=143 -> 144

### ISSUE THREAD

**Issue by itsfarseen** [keepassxreboot/keepassxc#7138]
## QT_FONT_DPI=143

![image](https://user-images.githubusercontent.com/40014472/142247275-289b8e3e-63cc-46f3-ac70-44c052898cde.png)

## QT_FONT_DPI=144

![2021-11-17_22-38](https://user-images.githubusercontent.com/40014472/142248010-1702e0fd-eeee-4b89-9dcd-b2f4bb8fd24c.png)

## Steps to Reproduce
1. Run QT_FONT_DPI=143 keepassxc
2. Run QT_FONT_DPI=144 keepassxc

## Expected Behavior
`QT_FONT_DPI=144 keepassxc` should only be slightly larger than when DPI is set to 143.

## Actual Behavior
It is too large.

## Context

I was investigating why only keepassxc among all the Qt apps I have (pcmanfm-qt, flameshot, sqlitebrowser) had a too large interface.  
It can also be worked around by setting `QT_SCALE_FACTOR=0.75` right before launching keepasxc.

KeePassXC - Version 2.6.6
Revision: 9c108b9

Qt 5.15.2
Debugging mode is disabled.

Operating system: Arch Linux
CPU architecture: x86_64
Kernel: linux 5.15.2-zen1-1-zen

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- libgcrypt 1.9.4-unknown

Operating System: Linux
Desktop Env: i3 with Gtk apps mostly
Windowing System: X11

**Comment by droidmonkey** [comment]
This is probably not a bug, we might be the only app in your list that is high dpi aware. Please report this to Qt, we don't do anything special when it comes to scaling.

**Comment by itsfarseen** [comment]
Fix for the above issue: https://stackoverflow.com/a/69126731/1770031

TLDR; Set the env var `QT_SCALE_FACTOR_ROUNDING_POLICY=PassThrough`

Qt rounds DPI values by default. That's why there is a sudden large jump between DPI values when your monitors have similar but slightly different DPIs and they happen to be near the middle of a round number.

So it went from 1x scaling to 2x scaling when my DPI changed from 143 to 144.

If I set this env var, I don't have the issue any more :)
